src/utils/Cypher.ts

```diff
@@ -26,19 +31,75 @@ export default class Cypher {
     return this.configuration;
   }
 
-  public static encrypt(data: string): string {
+  public static getCrypto(): CryptoSetting {
+    return this.cryptoSetting;
+  }
+
+  public static async detectConfigurationKey(tenantID: string): Promise<boolean> {
+
+    const configCryptoKey: string = Configuration.getCryptoConfig().key;
+    const keySettings = await SettingStorage.getCryptoSettings(tenantID);
+    let cryptoSettingToSave: CryptoSetting;
+
+    // Check if Crypto Settings exist
+    if (keySettings) {
+      // Detect new config Cypher key
+      if (keySettings.crypto?.key !== configCryptoKey) {
+        await SettingStorage.deleteSetting(tenantID, keySettings.id);
```

Comment:
It's probably not needed. The saveCryptoSettings() method is doing an update MongoDB request. 
If you remove it, it should work.

Reply:
Actually it does not work without it. It's not clear to me why, because I see that saveSettings() is using findOneAndUpdate(). But it blocks there if I don't delete first. The findOneAndUpdate() does not resolve.

Reply:
I was not passing the id of the setting, that's why it was throwing an error. Now it works without the deleteSetting. Thanks :)